lib/windshaft/renderers/mapnik/factory.js

```
@@ -289,20 +287,8 @@ MapnikFactory.prototype.mapConfigToMMLBuilderConfig = function(mapConfig, queryR
         gcols: []
     };
 
-    try {
-        filteredLayerIndexes = layersFilter(mapConfig, mapConfig.getLayers(), layerFilter);
-    } catch (err) {
-        // aliases mapnik, all,
-        filteredLayerIndexes = [];
-        for (var i = 0; i < mapConfig.getLayers().length; i++) {
-            filteredLayerIndexes.push(i);
-        }
-    }
-
-    filteredLayerIndexes.filter(function (layerIndex) {
-        return mapConfig.layerType(layerIndex) === 'mapnik';
```

This `filter` was wrong or it is expected to silently ignore non-supported layers?

---

I'm relying on the fact that `layersFilter` is already being used in the RendererFactory, but good point, we might need to keep that filtering.

I'm also realising that `var filteredLayerIndexes = layersFilter(mapConfig, layerFilter);` might throw an exception, and I'm not handling it here.

---

Although there is another `throw` within this function, so I guess we can keep the `layersFilter` call without a try/catch block.